### DISCUSSION TITLE
Tweaking help short/expanded behavior

### DISCUSSION THREAD

**Discussion by ozwaldorf** [pacak/bpaf#367] [Ideas]
Coming from clap, help is expanded by using the `long` option, not by repeating either the short or long flag. This makes a lot of sense for me and I would like to preserve this behavior as bpaf's default is a bit unfamiliar, and I'd imagine it will be to users as well. Perhaps an enum like `ExpandedHelpBehavior { Repeat, ShortLong, Always }` that can be passed to the command parser would work

**Comment by pacak** [comment]
As in different output for `foo -h` and `foo --help`? I'm not really familiar with clap internals apart from `cargo` and some other tools that happen to use it. `cargo --help` and `cargo -H` produces the same output.

**Reply by ozwaldorf** [reply]
Exactly, very similar to how bpaf shows the expanded help if you repeat the help flag `-hh` or `--help --help`

Clap (v4) will trim help at the first double newline, and `-h` would only show the one liner, but `--help` would show the full thing and not remove any docs. Long help flag also seems to set the `NextLineHelp` setting, which is a bit more space efficient. I believe the help subcommand also carries the same logic as the long flag.

Example using lutgen 0.10.1, which uses clap v4:

![image](https://github.com/pacak/bpaf/assets/8976745/67db1618-4e36-44c9-a65c-3391f2f5bd5e)
![image](https://github.com/pacak/bpaf/assets/8976745/d4ad89b7-878f-4408-be34-c142e61bbe9d)
